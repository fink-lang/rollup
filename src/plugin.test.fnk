{resolve} = import 'path'

{describe, it, expect, to_match_snapshot, to_throw} = import '@fink/jest/test.fnk'
{fink} = import './plugin.fnk'



describe fink, fn:
  it 'transforms files', fn:
    {transform} = fink _

    {code, map: {sources: _, ...source_map}, ...rest} = transform
      'foobar = 13'
      resolve './src/test.fnk'

    expect
      {code, map: source_map, ...rest}
      to_match_snapshot


  it 'ignores files', fn:
    {transform} = fink {include: './src/**/*.fnk'}

    expect
      transform 'foobar = 13', resolve './src/test.js'
      to_match_snapshot


  it 'throws parsing errors', fn:
    {transform} = fink {include: './src/**/*.fnk'}

    filename = resolve './src/test.fnk'

    expect
      fn: transform 'foobar = "foo', resolve './src/test.fnk'
      to_throw '${filename}:1:13
        1| foobar = "foo
                        ^

        Unexpected end of code.
        '


  it 'throws compile errors', fn:
    {transform} = fink {include: './src/**/*.fnk'}

    filename = resolve './src/test.fnk'

    expect
      fn: transform '123 = foo', resolve './src/test.fnk'
      to_throw '${filename}:1:0
        1| 123 = foo
           ^

        Unable to transform `assign =`.

        TypeError: Property left of AssignmentExpression expected node to be of a type ["LVal"] but instead got "NumericLiteral"
        '