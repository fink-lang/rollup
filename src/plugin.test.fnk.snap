// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fink ignores files 1`] = `null`;

exports[`fink transforms files 1`] = `
Object {
  "code": "export const foobar = 13;",
  "map": Object {
    "mappings": "AAAA,aAAA,MAAM,GAAG,EAAT",
    "names": Array [],
    "sourcesContent": Array [
      "foobar = 13",
    ],
    "version": 3,
  },
  "moduleSideEffects": false,
}
`;
